tests/testthat/test-type-factor.R

```diff
@@ -105,6 +105,37 @@ test_that("vec_ptype2() errors with malformed ordered factors", {
   expect_error(vec_ptype2(y, x, y_arg = "z"), "`z` is a corrupt ordered factor")
 })
 
+test_that("ordered factors with different levels are not compatible", {
+  expect_error(
+    vec_ptype2(ordered("a"), ordered("b")),
+    class = "vctrs_error_incompatible_type"
+  )
+  expect_error(
+    vec_ptype2(ordered("a"), ordered(c("a", "b"))),
+    class = "vctrs_error_incompatible_type"
+  )
+})
```

Comment:
Maybe also a `vec_cast(ordered("a"), ordered(c("a", "b")))` check to ensure that `vec_cast()` is doing the right thing now too?